internal/engine/experiment/tor/tor.go

```diff
@@ -330,75 +325,59 @@ func maybeScrubbingLogger(input model.Logger, kt keytarget) model.Logger {
 	return &scrubber.Logger{Logger: input}
 }
 
-func (rc *resultsCollector) defaultFlexibleConnect(
-	ctx context.Context, kt keytarget,
-) (tk oonitemplates.Results, err error) {
-	logger := maybeScrubbingLogger(rc.sess.Logger(), kt)
+// defaultFlexibleConnect is the default implementation of the
+// rc.flexibleConnect testable function.
+//
+// Arguments:
+//
+// - ctx is the context for timeout/cancellation;
+//
+// - kt contains information about the target.
+//
+// Returns:
+//
+// - tk is the measurement, which is always non nil because
+// the measurex "easy" API provides this guarantee;
+//
+// - failure is nil or an OONI failure string.
+func (rc *resultsCollector) defaultFlexibleConnect(ctx context.Context,
+	kt keytarget) (tk *measurex.ArchivalMeasurement, failure *string) {
+	mx := measurex.NewMeasurerWithDefaultSettings()
+	mx.Begin = rc.measurement.MeasurementStartTimeSaved
+	mx.Logger = maybeScrubbingLogger(rc.sess.Logger(), kt)
 	switch kt.target.Protocol {
 	case "dir_port":
-		url := url.URL{
+		URL := url.URL{
 			Host:   kt.target.Address,
 			Path:   "/tor/status-vote/current/consensus.z",
 			Scheme: "http",
 		}
 		const snapshotsize = 1 << 8 // no need to include all in report
-		r := oonitemplates.HTTPDo(ctx, oonitemplates.HTTPDoConfig{
-			Accept:                  httpheader.Accept(),
-			AcceptLanguage:          httpheader.AcceptLanguage(),
-			Beginning:               rc.measurement.MeasurementStartTimeSaved,
-			MaxEventsBodySnapSize:   snapshotsize,
-			MaxResponseBodySnapSize: snapshotsize,
-			Handler:                 netxlogger.NewHandler(logger),
-			Method:                  "GET",
-			URL:                     url.String(),
-			UserAgent:               httpheader.UserAgent(),
-		})
-		tk, err = r.TestKeys, r.Error
+		mx.HTTPMaxBodySnapshotSize = snapshotsize
+		const timeout = 15 * time.Second
+		return mx.EasyHTTPGET(ctx, timeout, URL.String())
 	case "or_port", "or_port_dirauth":
-		r := oonitemplates.TLSConnect(ctx, oonitemplates.TLSConnectConfig{
-			Address:            kt.target.Address,
-			Beginning:          rc.measurement.MeasurementStartTimeSaved,
-			InsecureSkipVerify: true,
-			Handler:            netxlogger.NewHandler(logger),
-		})
-		tk, err = r.TestKeys, r.Error
+		tlsConfig := measurex.NewEasyTLSConfig().InsecureSkipVerify(true)
+		return mx.EasyTLSConnectAndHandshake(ctx, kt.target.Address, tlsConfig)
 	case "obfs4":
-		r := oonitemplates.OBFS4Connect(ctx, oonitemplates.OBFS4ConnectConfig{
-			Address:      kt.target.Address,
-			Beginning:    rc.measurement.MeasurementStartTimeSaved,
-			Handler:      netxlogger.NewHandler(logger),
-			Params:       kt.target.Params,
-			StateBaseDir: rc.sess.TempDir(),
-		})
-		tk, err = r.TestKeys, r.Error
```

Comment:
The address is still passed below. The beginning and the logger are set into `mx`. The state directory and the params go directly into the "easy" API call, which also has an explicit timeout.